### ISSUE TITLE
Remove Copy bounds for some methods

### ISSUE THREAD

**Issue by elidupree** [darkstalker/array_ext#1]
With the fairly recent new syntax for safe pattern matching to move out of non-Copy arrays, it should be possible to relax some (all?) of the Copy bounds.

e.g. this can be used to implement map():
```
let [$($identifier),*] = self;
[$(f($identifier)),*]
```
and zip():
```
let [$($identifier),*] = self;
let [$($other_identifier),*] = other;
[$(f($identifier, $other_identifier)),*]
```

Although of course, this requires that the macros take a list of identifiers rather than just the list of indices.

**Comment by wolfiestyle** [comment]
Nice, didn't know that got implemented. But I guess the implementation would look super ugly. Maybe can use some recursive trick to generate identifier lists inside the macros.

**Comment by wolfiestyle** [comment]
fixed in 3e331a531d53daca9a433830635c45a15cfada99
